### ISSUE TITLE
Mapnik 3 config ok but unable to compile

### ISSUE THREAD

**Issue by flp-gpc** [mapnik/mapnik#3042]
I am unable to compile mapnik 3. All dependencies are correctly installed, below is the result of the ./configure, but when I try to make, I receive hundreds of errors... Maybe some versions of the dependencies are not compatibles ? I am on Ubuntu 14.04.

Welcome to Mapnik...

Configuring build environment...
SCons CONFIG found: 'config.py', variables will be inherited...
Configuring on Linux in *release mode*...
Checking for freetype-config... yes
Checking for dlfcn.h support ... yes
Checking if compiler (c++) supports -std=c++11 flag... (cached) yes
Checking for C library z... yes
Checking for C++ library icuuc... yes
Checking for ICU version >= 4.2... found: icu 52.1
(cached) Checking for C++ library harfbuzz... yes
Checking for HarfBuzz version >= 0.9.34... found: HarfBuzz 1.0.2
(cached) Searching for boost libs and headers... (cached)
Using boost lib dir: /usr/lib
Using boost include dir: /usr/include
Checking for C++ header file boost/version.hpp... yes
Checking for Boost version >= 1.47... yes
Found boost lib version... 1_55
Checking for C++ library boost_system... yes
Checking for C++ library boost_filesystem... yes
Checking for C++ library boost_regex... yes
Checking for C++ library boost_program_options... yes
Checking for C++ library boost_thread... yes
Checking whether Boost was compiled with C++11 scoped enums ... no
Checking if boost_regex was built with ICU unicode support... (cached) yes
Checking for C library jpeg... yes
Checking for C library proj... yes
Checking for C library png... yes
Checking for C library webp... yes
Checking for C library tiff... yes
Checking for pkg-config... yes
Checking for requested plugins dependencies...
Checking for gdal-config --libs... yes
Checking for gdal-config --cflags... yes
Checking for name of gdal library... gdal
Checking for C++ library gdal... yes
Checking if gdal is ogr enabled... yes
Checking for gdal-config --libs... yes
Checking for gdal-config --cflags... yes
Checking for name of ogr library... gdal
Checking for C++ library gdal... yes
Checking for pg_config... yes
Checking for pg_config... yes
Checking for C library sqlite3... yes
Checking if SQLite supports RTREE... (cached) yes
Checking for cairo... yes
Checking for cairo lib and include paths...  yes
Checking for cairo freetype font support ... yes

All Required dependencies found!

**Comment by flippmoke** [comment]
Can I see some of your compile errors?

**Comment by flemmens** [comment]
Of course, here are some lines. I receive thousands like this so I am not sure what to select to be helpful. I can redirect all the errors in a file if needed.

In file included from src/renderer_common/process_group_symbolizer.cpp:25:0:
include/mapnik/renderer_common/process_markers_symbolizer.hpp: In member function âvoid mapnik::render_marker_symbolizer_visitor<VD, RD, RendererType, ContextType>::operator()(const mapnik::marker_svg&)â:
include/mapnik/renderer_common/process_markers_symbolizer.hpp:94:32: warning: declaration of âattributesâ shadows a global declaration [-Wshadow]
             svg_attribute_type attributes;
                                ^
In file included from include/mapnik/renderer_common.hpp:30:0,
                 from include/mapnik/renderer_common/process_group_symbolizer.hpp:31,
                 from src/renderer_common/process_group_symbolizer.cpp:24:
include/mapnik/attribute.hpp:73:58: note: shadowed declaration is here
 using attributes = std::unordered_map<std::string, value>;
                                                          ^
In file included from src/renderer_common/process_group_symbolizer.cpp:25:0:
include/mapnik/renderer_common/process_markers_symbolizer.hpp:142:32: warning: declaration of âattributesâ shadows a global declaration [-Wshadow]
             svg_attribute_type attributes;
                                ^
In file included from include/mapnik/renderer_common.hpp:30:0,
                 from include/mapnik/renderer_common/process_group_symbolizer.hpp:31,
                 from src/renderer_common/process_group_symbolizer.cpp:24:
include/mapnik/attribute.hpp:73:58: note: shadowed declaration is here
 using attributes = std::unordered_map<std::string, value>;
                                                          ^
src/renderer_common/process_group_symbolizer.cpp: In member function âvoid mapnik::render_thunk_extractor::operator()(const mapnik::markers_symbolizer&) constâ:
src/renderer_common/process_group_symbolizer.cpp:166:52: warning: declaration of âusing context_type = class std::tuple<std::list<std::unique_ptr<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk> > >, std::allocator<std::unique_ptr<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk> > > > >&>â shadows a global declaration [-Wshadow]
     using context_type = decltype(renderer_context);
    
D = mapnik::detail::raster_marker_thunk_dispatch<mapnik::label_collision_detector4, std::tuple<std::list<std::unique_ptr<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk> > >, std::allocator<std::unique_ptr<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk> > > > >&> >; RendererType = mapnik::virtual_renderer_common; ContextType = std::tuple<std::list<std::unique_ptr<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk> > >, std::allocator<std::unique_ptr<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk> > > > >&>]â
src/renderer_common/process_group_symbolizer.cpp:171:83:   required from here
include/mapnik/transform_path_adapter.hpp:66:14: warning: declaration of âgeomâ shadows a member of 'this' [-Wshadow]
     explicit transform_path_adapter(Geometry & geom)
              ^

D = mapnik::detail::raster_marker_thunk_dispatch<mapnik::label_collision_detector4, std::tuple<std::list<std::unique_ptr<mapnik::util::variant<mapnik::vector_markerrender_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::aster_marker_render_thunk, mapnik::text_render_thunk> > >, std::allocator<std::unique_ptr<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_mrker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapik::text_render_thunk> > > > >&> >; RendererType = mapnik::virtual_renderer_common; ContextType = std::tuple<std::list<std::unique_ptr<mapnik::util::variant<mapnik:vector_marker_render_thunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_tunk, mapnik::raster_marker_render_thunk, mapnik::text_render_thunk> > >, std::allocator<std::unique_ptr<mapnik::util::variant<mapnik::vector_marker_render_thunk, manik::raster_marker_render_thunk, mapnik::text_render_thunk>, std::default_delete<mapnik::util::variant<mapnik::vector_marker_render_thunk, mapnik::raster_marker_rener_thunk, mapnik::text_render_thunk> > > > >&>]â
src/renderer_common/process_group_symbolizer.cpp:171:83:   required from here
deps/agg/include/agg_conv_clip_polyline.h:45:14: warning: declaration of ây2â shadows a member of 'this' [-Wshadow]
         void clip_box(double x1, double y1, double x2, double y2)
              ^
deps/agg/include/agg_conv_clip_polyline.h:45:14: warning: declaration of âx2â shadows a member of 'this' [-Wshadow]
deps/agg/include/agg_conv_clip_polyline.h:45:14: warning: declaration of ây1â shadows a member of 'this' [-Wshadow]
deps/agg/include/agg_conv_clip_polyline.h:45:14: warning: declaration of âx1â shadows a member of 'this' [-Wshadow]

**Comment by flippmoke** [comment]
Can you try compiling with `-Wshadow` turned off?

**Comment by springmeyer** [comment]
@flippmoke `-Wshadow` is always on via https://github.com/mapnik/mapnik/blob/master/SConstruct#L1772. The problem is not evident in the logs @flemmens shared. All we can see is warnings and not the actual error. @flemmens - so we can't help you unless you provide the entire terminal output so we can find the actual error. Please do that and paste it to https://gist.github.com/ and share the link.

**Comment by flemmens** [comment]
Thank you for your help. Actually I was finally able to compile. It seems there was only warnings but as so many messages appears on my screen (literally thousands) I was unable to determine if there was actual errors inside all that mess. I just make a quick rendering test and it worked, so it seems everything has correctly been compiled.

However, I don't know if it possible to update the makefile to avoid these warning messages ? I used a fresh install Ubuntu 14.04 on a new server with a very standard config. All other sources I have compiled were flawless.

Thanks for the support anyway and for this great tool.
